### PR TITLE
Add countdown to showing command palette `no matches` message

### LINKED ISSUE #3278: Command palette flicker when there are no results
When there are no results, each keypress produces a flicker where the "No matches found" message is shown and hidden. I think we should be able to avoid this, so the "No matches found" message persists.

https://github.com/Textualize/textual/assets/554369/98e93373-e7e1-4f95-84e7-0f193684cc7c

### PR DESCRIPTION
**Pull request by guy-av**
The countdown protects from spamming the message and helping the palette feel smoother when there are no matches.

- [x] Docstrings on all new or modified functions / classes
- [x] Change `test_no_results` to wait for the countdown

Resolves #3278

### CODE REVIEW

**Comment by guy-av** [comment]
This PR tries to follow suggested changes noted in [this comment](https://github.com/Textualize/textual/issues/3278#issuecomment-1713875625).

File: src/textual/command.py [hunk 1]
```diff
@@ -559,6 +561,38 @@ def _become_busy() -> None:
 
         self._busy_timer = self.set_timer(self._BUSY_COUNTDOWN, _become_busy)
 
+    def _stop_no_matches_countdown(self) -> None:
```
Comment (davep) [inline src/textual/command.py:564]: Could we get a docstring for this please?

File: src/textual/command.py [hunk 2]
```diff
@@ -559,6 +561,38 @@ def _become_busy() -> None:
...
+        if self._no_matches_timer is not None:
+            self._no_matches_timer.stop()
+            self._no_matches_timer = None
+
+    _NO_MATCHES_COUNTDOWN: Final[float] = 0.5
+    """How many seconds to wait before showing 'No matches found'"""
```
Comment (davep) [inline src/textual/command.py:570]:
```suggestion
    """How many seconds to wait before showing 'No matches found'."""
```

File: src/textual/command.py [hunk 3]
```diff
@@ -559,6 +561,38 @@ def _become_busy() -> None:
...
+
+    def _start_no_matches_countdown(self, command_list: CommandList) -> None:
```
Comment (davep) [inline src/textual/command.py:572]: What's the thinking about having the `CommandList` instance be an argument here? The `CommandList` that is part of the command palette should be queryable.

**Review by davep — COMMENTED** [review]
Looking good; thanks. Couple of very small tweaks and one question.

Also, very small general observation (which really isn't a problem, *very* much appreciate the contribution, but worth a mention in regards to FOSS contributions in general): this issue was already assigned so it would be nice to check with the assignee first if it's okay to jump in.

**Comment by guy-av** [comment]
> Looking good; thanks. Couple of very small tweaks and one question.
> 
> 
> 
> Also, very small general observation (which really isn't a problem, *very* much appreciate the contribution, but worth a mention in regards to FOSS contributions in general): this issue was already assigned so it would be nice to check with the assignee first if it's okay to jump in.

Sure thing!

File: src/textual/command.py (hunk 3 above)
Comment (guy-av) [reply src/textual/command.py:572]: You're right!
Nice catch

Comment (guy-av) [reply src/textual/command.py]: The argument has been removed and the `CommandList` is queried instead

**Review by davep — APPROVED** [review]
Looks good to me! Thanks again.

I've also added an entry to the CHANGELOG.

**Pull request merged**